src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -21,6 +21,7 @@ internal class RemoteCredentialsDecorator : ICredentialsDecorator {
         private readonly AsyncReaderWriterLock _lock;
         private bool _credentialsAreValid;
         private readonly string _workspaceName;
+        private object _credentialsLock;
```

Comment:
What I think we should do is make _credentials volatile, implement GetCredential like this:
```cs
public NetworkCredential GetCredential(Uri uri, string authType) {
    var credentials = _credentials;
    return credentials != null ? new NetworkCredential(credentials.UserName, credentials.Password) : new NetworkCredential();
}
```

and don't add any new locks.